src/js/common/helpers/login-helpers.js

```diff
@@ -17,14 +17,30 @@ export function handleLogin() {
 }
 
 export function handleVerify() {
-  this.serverRequest = fetch(`${environment.API_URL}/v0/sessions/new?level=3`, {
+  window.dataLayer.push({
+    event: 'verify-link-clicked',
+  });
+  const urlPromise = fetch(`${environment.API_URL}/v0/sessions/new?level=3`, {
     method: 'GET',
   }).then(response => {
     return response.json();
   }).then(json => {
-    const myVerifyUrl = json.authenticate_via_get;
-    const receiver = window.open(myVerifyUrl, '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
-    receiver.focus();
+    return json.authenticate_via_get;
+  });
+
+  // Open the window outside of the promise callback to avoid the browser blocking it.
+  // In case the url fetch fails, we should fix this to behave more gracefully than closing the window.
+  const receiver = window.open('', '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+  urlPromise.then(myVerifyUrl => {
```

Comment:
I'm a little confused what `myVerifyUrl` is in this case. A new function? It looks like we removed declaring the variable `myVerifyUrl` and setting it to `json.authenticate_via_get` from the old code, so it's no longer defined up until this point, right?